dali/pipeline/executor/executor.cc

```diff
@@ -525,13 +525,14 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunHelper(OpNode &op_node, Workspac
     }
   }
 
-  // If it is the end of an epoch, create a checkpoint.
+  // Create a checkpoint.
   // The checkpoint corresponds to the state between the iteration `iteration_id`
   // and `iteration_id + 1`.
-  // After consuming all the outputs from the epoch, GetCurrentCheckpoint is going to
+  // After consuming all the outputs from the iteration, GetCurrentCheckpoint is going to
   // return the created checkpoint.
-  if (checkpointing_ && (iteration_id + 1) % checkpointing_epoch_size_ == 0)
```

Comment:
It seems we could get rid of the `checkpointing_epoch_size_` completely at this point, right? Along with the check that all the readers in the pipeline agree on the epoch size.

Reply:
That's right, removed.